proposed/2022/cpm-dotnet-cli-add-package-support.md

```diff
@@ -4,218 +4,148 @@
+| 2 | ❌ | ❌ | ❌ | ✔️ | Add `PackageReference` to the project file. Add `PackageVersion` to the `Directory.Packages.Props` file. Use version specified in the commandline. | ✔️ |
+| 3 | ❌ | ❌ | ✔️ | ❌ |  Add `PackageReference` to the project file. No changes to the `Directory.Packages.Props` file. Basically we are reusing the version defined centrally for this package. | ✔️ |
+| 4 | ❌ | ❌ | ✔️ | ✔️ | Add `PackageReference` to the project file. Update `PackageVersion` in the `Directory.Packages.Props` file with the version specified in the commandline.  | ✔️ |
+| 5 | ❌ | ✔️ | ❌ | ❌ | Not a valid scenario because a `VersionOverride` can't exist without `PackageReference`. | ❌ |
+| 6 | ❌ | ✔️ | ❌ | ✔️ | Not a valid scenario because a `VersionOverride` can't exist without `PackageReference`. | ❌ |
+| 7 | ❌ | ✔️ | ✔️ | ❌ | Not a valid scenario because a `VersionOverride` can't exist without `PackageReference`. | ❌ |
+| 8 | ❌ | ✔️ | ✔️ | ✔️ | Not a valid scenario because a `VersionOverride` can't exist without `PackageReference`. | ❌ |
+| 9 | ✔️ | ❌ | ❌ | ❌ | Emit an error. (The other considered option was: Remove `Version` from `PackageReference`, Add `PackageVersion` to the `Directory.Packages.Props` file. Use `Version` from `PackageReference` if it exists otherwise use latest version from the package sources. However, this might require more user research.) | ✔️ |
+| 10 | ✔️ | ❌ | ❌ | ✔️ | Emit an error. (The other considered option was: Remove `Version` from `PackageReference`, Add `PackageVersion` to the `Directory.Packages.Props` file. Use `Version` passed in the commandline.  However, this might require more user research.) | ✔️ |
+| 11 | ✔️ | ❌ | ✔️ | ❌ | No-op. (The other considered option was: Update `PackageVersion` in the `Directory.Packages.Props` file,  use latest version from the package sources. However, we decided it would be best to only update a version when the user explicitly passes in the commandline argument.) | ✔️ |
+| 12 | ✔️ | ❌ | ✔️ | ✔️ | Update `PackageVersion` in the `Directory.Packages.Props` file, use version specified in the commandline. | ✔️ |
+| 13 | ✔️ | ✔️ |❌  | ❌ | No-op. | ✔️ |
+| 14 | ✔️ | ✔️ | ❌ | ✔️ | Update `VersionOverride` in the existing `PackageReference` item, use version specified in the commandline. | ✔️ |
+| 15 | ✔️ | ✔️ | ✔️ | ❌ | No-op. | ✔️ |
+| 16 | ✔️ | ✔️ | ✔️ | ✔️ | Update `VersionOverride` in the existing `PackageReference` item, use version specified in the commandline. | ✔️ |
+
+### Technical explanation
+
+NuGet restore operation generates `{projectName}.nuget.dgspec.json` file that file that maintains a project's top-level dependencies along with other metadata for `PackageReference` style projects. 
+- `dgspec.json` file will have `centralPackageVersionsManagementEnabled` property set to `true` for projects onboarded onto CPM.
```

Comment:
I mentioned this in previous PRs. This should be removed.

This is incorrect. You're not going to be using the dgspec.json generated by restore.

You're using a dgspec json explicitly generated by the add package command.

Reply:
Noted, the wording here has been updated.